tools/keyboard/scancodes_LCID_files_generator.cpp

```diff
@@ -0,0 +1,393 @@
+        ofs << "  // key : extended unicode, value : extended scancode + unicode [8 + 16 bits]\n"
+            << "  inline const std::unordered_map<std::uint16_t, std::pair<std::uint8_t, std::uint16_t>> EXTENDED_SCANCODES_TABLE\n"
+            << "  {\n";
+
+        std::unordered_set<std::uint16_t> extended_unicode_occurences;
+        
+        for (std::uint8_t i = 0; i < layout.nbDeadkeys; ++i)
+        {
+            for (std::uint8_t j = 0; j < layout.deadkeys[i].nbSecondKeys; ++j)
+            {
+                std::uint16_t extended_unicode =
+                    layout.deadkeys[i].secondKeys[j].modifiedKey;
+
+                if (extended_unicode_occurences.find(extended_unicode)
+                    == extended_unicode_occurences.cend())
+                {
+                    ofs << "    { ";
+                    code_hex16_stream_fmt(extended_unicode);
+                    ofs << ", { ";
+                
+                    std::uint8_t extended_scancode =
+                        layout.deadkeys[i].extendedKeyCode;
+                    std::uint16_t unicode =
+                        layout.deadkeys[i].secondKeys[j].secondKey;
+                
+                    code_hex8_stream_fmt(extended_scancode);
+                    ofs << ", ";
+                    code_hex16_stream_fmt(unicode);
+                    ofs << " } },\n";
+                    
+                    extended_unicode_occurences.emplace(extended_unicode);
+                }
+            }
+        }
+        
+        ofs << "  };\n"
+    
+            << "}\n";
+    }
+
+    void write_header_scancodes_LCID_file(std::ofstream& ofs)
+    {   
+        ofs << "#pragma once\n\n"
+            << "# include <cstdint>\n"
+            << "# include <unordered_map>\n\n";
+    }
+
+    void generate_scancodes_LCID_file(const Keylayout& layout,
+                                      LCIDs_t& lcids)
+    {
+        constexpr auto FILENAME_PREFIX = "scancodes_"_zv;
+        constexpr auto FILENAME_SUFFIX = ".hpp"_zv;
+    
+        std::string LCID = number_to_hex_string(layout.LCID, "x");
+        std::string filename;
+        std::string file_path;
+
+        filename.reserve(FILENAME_PREFIX.size()
+                         + LCID.size()
+                         + FILENAME_SUFFIX.size());
+        filename
+            .append(FILENAME_PREFIX.c_str(), FILENAME_PREFIX.size())
+            .append(LCID)
+            .append(FILENAME_SUFFIX.c_str(), FILENAME_SUFFIX.size());
+        
+        file_path.reserve(GEN_DIR_PATH.size()
+                          + filename.size());
+        file_path
+            .append(GEN_DIR_PATH.c_str(), GEN_DIR_PATH.size())
+            .append(filename);
+        
+    
+        std::ofstream ofs;
+
+        ofs.exceptions(std::ofstream::failbit | std::ofstream::badbit);
+        try
+        {
+            ofs.open(file_path.c_str(),
+                     std::ios_base::out | std::ios_base::trunc);
```

Comment:
`.c_str()` is superfluous, same for `std::ios_base::out | std::ios_base::trunc`.

```cpp
ofs.open(file_path);
```